tests/core/test_log.py

```diff
@@ -73,6 +71,19 @@ def test_log_exact_match_raises_exception_with_invalid_args_kwargs(
         Log(ping.events.Ping, *args, **kwargs).exact_match(receipt)
 
 
+@pytest.mark.parametrize(
+    "kwargs",
+    (
+        ({"invalid": b"1"}),
+        ({"first": b"1", "invalid": b"2"}),
+    ),
+)
+def test_invalid_keywords_raise_exception_on_log_instantiation(ping_setup, kwargs):
+    ping, receipt = ping_setup
+    with pytest.raises(LogError):
+        Log(ping.events.Ping, **kwargs)
```

Comment:
@pipermerriam 
> I would expect this to raise a TypeError since the arguments are not valid for the event's ABI, and for the exception to be raised on instantiation of the Log class.

It's raising a `TypeError` it's just that I catch that and raise a `LogError` instead in `Log`'s [constructor](https://github.com/ethereum/pytest-ethereum/blob/master/pytest_ethereum/testing.py#L25) - unless you're thinking I should just let the `TypeError` propagate through?

Reply:
Often we still still use things like `TypeError/AttributeError` when the cause of the exception fits the standard pattern and it isn't really a library specific error like in this case where they are just instantiating the class incorrectly.